learn/advanced/asynchronous_operations.md

```diff
@@ -118,17 +136,83 @@ Had the task failed, the response would have included an `error` object:
 }
 ```
 
+If the task was canceled while it was `enqueued` or `processing`, it would have the `canceled` status:
+
+```json
+{
+  "uid": 1,
+  "indexUid": null,
+  "status": "canceled",
+  "type": "taskCancelation",
+  "canceledBy": 5,
+  "details": {
+    "matchedTasks": 1,
+    "canceledTasks": 1,
+    "originalQuery": "uid=1"
+  },
+  "error": null,
+  "duration": "PT1S",
+  "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+  "startedAt": "2021-08-10T14:29:18.000000Z",
+  "finishedAt": "2021-08-10T14:29:19.000000Z"
+}
+```
+
+If the task is deleted after it was processed (`succeeded`, `failed`, or `canceled`), Meilisearch would return a [`task_not_found`](/reference/errors/error_codes.md#task-not-found) error.
+
 ### Filtering tasks
 
-You can filter tasks based on `status`, `type`, or `indexUid`. For example, the following command returns all tasks belonging to the index `movies`. Note that the `indexUid` is case-sensitive:
+You can filter tasks based on `uid`, `status`, `type`, `indexUid`, `canceledBy`, or date.
+
+#### Filter by `uid`
+
+The following code sample returns tasks with `uid`s `5`, `10`, and `13`:
+
+<CodeSamples id="async_guide_filter_by_id_1" />
+
+#### Filter by `status`
+
+The following code sample returns tasks with the `failed` and `canceled` statuses:
 
-<CodeSamples id="get_all_tasks_filtering_1" />
+<CodeSamples id="async_guide_filter_by_status_1" />
 
-Use the ampersand character `&` to combine filters, equivalent to a logical `AND`. Use the comma character `,` to add multiple filter values for a single field.
+#### Filter by `type`
 
-For example, the following command would return all `documentAdditionOrUpdate` tasks that either `succeeded` or `failed`:
+The following code sample returns `dumpCreation` and `indexSwap` tasks:
 
-<CodeSamples id="get_all_tasks_filtering_2" />
+<CodeSamples id="async_guide_filter_by_type_1" />
+
+#### Filter by `indexUid`
+
+The following command returns all tasks belonging to the index `movies`. Note that the `indexUid` is case-sensitive:
+
+<CodeSamples id="async_guide_filter_by_index_uid_1" />
+
+#### Filter by `canceledBy`
+
+#### Filter by date
+
+You can filter tasks using `beforeXAt` and `afterXAt` with the `enqueuedAt`, `startedAt`, and `finishedAt` fields. This filter accepts the [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) format. The following date syntaxes are valid:
```

Comment:
I'm not sure if saying `beforeXat` and `afterXAt` helps the user to understand that X is a placeholder for `enqueuedAt`, `startedAt` and `finishedAt`